staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller_test.go

```diff
@@ -0,0 +1,167 @@
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: corev1.ResourceQuotaSpec{
+			Hard: corev1.ResourceList{
+				corev1.ResourcePods: *resource.NewQuantity(2, resource.BinarySI),
+			},
+			Scopes: []corev1.ResourceQuotaScope{
+				scope,
+			},
+		},
+		Status: corev1.ResourceQuotaStatus{
+			Hard: corev1.ResourceList{
+				corev1.ResourcePods: *resource.NewQuantity(2, resource.BinarySI),
+			},
+			Used: corev1.ResourceList{
+				corev1.ResourcePods: *resource.NewQuantity(used, resource.BinarySI),
+			},
+		},
+	}
+}
+
+type testPodLister struct {
+	pods []*corev1.Pod
+}
+
+// List will return all objects across namespaces
+func (q *testPodLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	for i := range q.pods {
+		ret = append(ret, q.pods[i])
+	}
+	return ret, nil
+}
+
+// Get will attempt to retrieve assuming that name==key
+func (q *testPodLister) Get(name string) (runtime.Object, error) {
+	for _, curr := range q.pods {
+		if curr.Name == name {
+			return curr, nil
+		}
+	}
+	return nil, apierrors.NewNotFound(corev1.Resource("resourcequota"), name)
+}
+
+// ByNamespace will give you a GenericNamespaceLister for one namespace
+func (q *testPodLister) ByNamespace(namespace string) cache.GenericNamespaceLister {
+	namespacedPods := []*corev1.Pod{}
+	for i, curr := range q.pods {
+		if curr.Namespace == namespace {
+			namespacedPods = append(namespacedPods, q.pods[i])
+		}
+	}
+	return &testPodLister{namespacedPods}
+}
+
+func TestCheckRequest(t *testing.T) {
+	firstTestStartingPodState := []*corev1.Pod{
+		newTerminatingPod("foo", "one"),
+	}
+	podEvaluator := core.NewPodEvaluator(func(schema.GroupVersionResource) (cache.GenericLister, error) {
+		return &testPodLister{firstTestStartingPodState}, nil
+	}, testing2.NewFakeClock(time.Now()))
+
+	type args struct {
+		quotas    []corev1.ResourceQuota
+		a         admission.Attributes
+		evaluator v1.Evaluator
+		limited   []resourcequotaapi.LimitedResource
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []corev1.ResourceQuota
+		wantErr bool
+	}{
+		{
+			name: "standard",
+			args: args{
+				quotas: []corev1.ResourceQuota{
+					*newScopedQuota("foo", "terminating", corev1.ResourceQuotaScopeTerminating, 1),
+					*newScopedQuota("foo", "nonterminating", corev1.ResourceQuotaScopeNotTerminating, 1),
+				},
+				a: admission.NewAttributesRecord(
+					newTerminatingPod("foo", "one"),
+					newNonTerminatingPod("foo", "one"),
+					schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"},
+					"foo",
+					"one",
+					schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"},
+					"",
+					admission.Update,
+					nil,
+					false,
+					nil,
+				),
+				evaluator: podEvaluator,
+				limited:   nil,
+			},
+			want: []corev1.ResourceQuota{
+				*newScopedQuota("foo", "terminating", corev1.ResourceQuotaScopeTerminating, 2),
+				*newScopedQuota("foo", "nonterminating", corev1.ResourceQuotaScopeTerminating, 0),
```

Comment:
@deads2k According to https://github.com/kubernetes/kubernetes/pull/130060#issuecomment-2667389066, the expected result should be `*newScopedQuota("foo", "nonterminating", corev1.ResourceQuotaScopeTerminating, 1)`